docs/study-def-tricks.md

```diff
@@ -0,0 +1,318 @@
+
+```py
+admission_date1 = patients.with_these_clinical_events(
+                    codelist,
+                    returning="date",
+                    on_or_after="index_date",
+                    date_format="YYYY-MM-DD",
+                    find_first_match_in_period=True,
+		    return_expectations = {...}
+                ),
+				
+admission_date2 = patients.with_these_clinical_events(
+                    codelist,
+                    returning="date",
+                    on_or_after="admission_date1 + 1 day",
+                    date_format="YYYY-MM-DD",
+                    find_first_match_in_period=True,
+                    return_expectations = {...}
+                ),
+				
+...
+
+admission_date5 = patients.with_these_clinical_events(
+                    codelist,
+                    returning="date",
+                    on_or_after="admission_date4 + 1 day",
+                    date_format="YYYY-MM-DD",
+                    find_first_match_in_period=True,
+                    return_expectations = {...}
+                ),
+```
+This is ok for 5 events, but if you need more it becomes quite cumbersome. An alternative is to wrap the variable signature in a function and loop that signature over the number of events required:
+
+
+```py
+def with_these_clinical_events_date_X(name, codelist, index_date, n, return_expectations):
+    
+    def var_signature(name, codelist, on_or_after, return_expectations):
+        return {
+            name: patients.with_these_clinical_events(
+                    codelist,
+                    returning="date",
+                    on_or_after=on_or_after,
+                    date_format="YYYY-MM-DD",
+                    find_first_match_in_period=True,
+                    return_expectations=return_expectations
+	    ),
+        }
+    variables = var_signature(f"{name}_1", codelist, index_date, return_expectations)
+    for i in range(2, n+1):
+        variables.update(var_signature(f"{name}_{i}", codelist, f"{name}_{i-1} + 1 day", return_expectations))
+    return variables
+```
+
+This can then be used inside a study definition for example as follows:
+
+
+```py
+study = StudyDefinition(
+
+	...
+	
+	**with_these_clinical_events_date_X(
+		name = "probable_covid_date", 
+		codelist = probable_codes, 
+		returning="date",
+		on_or_after = "index_date",
+		n = 5, 
+		return_expectations = {
+			"date": {"earliest": "2020-05-01", "latest": "2021-06-01"},
+			"rate": "uniform",
+			"incidence": 0.05,
+		},
+	),
+	
+	...
+)
+```
+
+This can be used for any variable where you want to return information for a series of consecutive events. 
+
+The will create `n` columns in the dataset. A few things to be aware of:
+
+* This does not make for efficient SQL queries. It is creating a new query for each event, rather than a query for each set of events. This will slow down extraction time considerably.
```

Comment:
"This does not make for efficient SQL queries" <- the user doesn't necessarily know the relationship between study definitions and SQL. Is there a way to reword that's simply about performance in general?